docs/src/models/turbulence_closures.md

```diff
@@ -33,71 +35,274 @@ julia> vertical_closure = VerticalScalarDiffusivity(ν=1e-3, κ=2e-3)
+ScalarBiharmonicDiffusivity{HorizontalFormulation}(ν=1.0e8, κ=1.0e8)
+```
+
+#### Discrete diffusivity functions
+
+Viscosity or diffusivity can be specified as a discrete function of grid indices and model state.
+This is useful for resolution-dependent coefficients.
+For example, we can set ``\nu = A^2 / \lambda``, where ``A`` is the grid cell area and ``\lambda``
+is a damping timescale:
+
+```jldoctest closures
+julia> using Oceananigans, Oceananigans.Units
+
+julia> @inline νhb(i, j, k, grid, ℓx, ℓy, ℓz, clock, fields, λ) =
+           Oceananigans.Operators.Az(i, j, k, grid, ℓx, ℓy, ℓz)^2 / λ
+νhb (generic function with 1 method)
+
+julia> closure = HorizontalScalarBiharmonicDiffusivity(ν=νhb, discrete_form=true, parameters=15days)
+ScalarBiharmonicDiffusivity{HorizontalFormulation}(ν=Oceananigans.TurbulenceClosures.DiscreteDiffusionFunction{Nothing, Nothing, Nothing, Float64, typeof(νhb)}, κ=0.0)
+```
+
+The `ℓx`, `ℓy`, `ℓz` arguments indicate the grid location (`Center()` or `Face()`) where
+the diffusivity is evaluated.
 
+### Tupled closures
 
-## Smagorinsky-Lilly
+Multiple closures can be combined in a tuple. All closure contributions are summed:
 
-To use the default Smagorinsky-Lilly LES closure, we write
+```jldoctest
+julia> using Oceananigans
+
+julia> closure = (HorizontalScalarDiffusivity(ν=1e-3), VerticalScalarDiffusivity(ν=1e-4))
+(HorizontalScalarDiffusivity{ExplicitTimeDiscretization}(ν=0.001, κ=0.0), VerticalScalarDiffusivity{ExplicitTimeDiscretization}(ν=0.0001, κ=0.0))
+```
+
+See the [hydrostatic modeling example](@ref hydrostatic-example) below for a more complex tuple.
+
+## Closures for large eddy simulation
+
+These closures compute eddy viscosity and diffusivity from resolved flow features
+for large eddy simulation (LES).
+
+### `SmagorinskyLilly`
+
+[`SmagorinskyLilly`](@ref) computes an eddy viscosity from the resolved strain rate
+with a buoyancy-based stability correction:
 
 ```jldoctest
-julia> using Oceananigans.TurbulenceClosures
+julia> using Oceananigans
 
-julia> closure = SmagorinskyLilly()
+julia> SmagorinskyLilly()
 Smagorinsky closure with
 ├── coefficient = LillyCoefficient(smagorinsky = 0.16, reduction_factor = 1.0)
 └── Pr = 1.0
 ```
 
-The parameters `C`, `Cb`, and `Pr` may alternatively be specified explicitly.
-For more details see [`SmagorinskyLilly`](@ref).
+### `DynamicSmagorinsky`
+
+[`DynamicSmagorinsky`](@ref) dynamically computes the Smagorinsky coefficient from the resolved flow.
+Two averaging methods are available:
+
+**Lagrangian averaging** (default): averages along fluid parcel trajectories:
+
+```jldoctest
+julia> using Oceananigans
+
+julia> DynamicSmagorinsky()
+DynamicSmagorinsky{Float64}:
+├── averaging = Oceananigans.TurbulenceClosures.Smagorinskys.LagrangianAveraging()
+├── schedule = IterationInterval(1, 0)
+├── Pr = 1.0
+└── minimum_numerator = 1.0e-32
+```
+
+**Directional averaging**: averages along specified dimensions, useful for flows with
+homogeneous directions:
```

Comment:
Can we add a reference to Bou-Zeid (2005) somewhere around here? I guess same goes for other closures